pkg/aws/eni/types/types.go

```diff
@@ -98,6 +98,12 @@ type ENISpec struct {
 	// +kubebuilder:validation:Optional
 	SubnetTags map[string]string `json:"subnet-tags,omitempty"`
 
+	// InitialSubnetID is the subnet of the primary ENI the instance was brought
+	// up with. It is used as a sensible default subnet to create ENIs in.
+	//
+	// +kubebuilder:validation:Optional
+	InitialSubnetID string `json:"initial-subnet-id,omitempty"`
```

Comment:
nit: Maybe we should call this `NodeSubnetID`, to make it a bit more clear that it is the subnet of the node IP.